Create placeholder files and directories based on hca

jbalboni opened the pull request:
Addresses #2716.

This has some stripped down and placeholder files/directories based on the hca folder. It looks like there's more we could reuse from hca, but since I'm not sure exactly how much, I just brought over the minimum.

I haven't figure out how to view this app in vets-website yet, but it builds in Webpack so I'm putting it up for review in the meantime.

awong-dev committed:
Pull HCA code into temporary migration directory.

awong-dev committed:
Pull HCA code into temporary migration directory.

awong-dev committed:
Move source and most tests into right directories.

This does NOT build yet.

awong-dev committed:
Move source and most tests into right directories.

This does NOT build yet.

awong-dev committed:
Make HCA Build. Styles and tests still broken.

But this is much closer!

awong-dev committed:
Make HCA Build. Styles and tests still broken.

But this is much closer!

awong-dev committed:
Use the right entry.

awong-dev committed:
Use the right entry.

awong-dev committed:
Add back in hca SCSS.

awong-dev committed:
Add back in hca SCSS.

awong-dev committed:
Dedup sytles between _shame and hca.

Also reorder the variable loading. Importing our uswds variables before
importing all the uswds templates seems to undo our settings for the
usa-max-width.

awong-dev committed:
Dedup sytles between _shame and hca.

Also reorder the variable loading. Importing our uswds variables before
importing all the uswds templates seems to undo our settings for the
usa-max-width.

crwallace committed:
import ha styles into bundle

crwallace committed:
import ha styles into bundle

jbalboni committed:
Create placeholder files and directories based on hca

jbalboni committed:
Create placeholder files and directories based on hca

markolson commented:
`options-for-select.js` looks like a _prime_ candidate for pulling into a common shared lib in this larger app/repo

jbalboni commented:
@markolson Yeah, I think so.

jbalboni committed:
Merge branch 'hca' into edu-benefits-setup-jsb

jbalboni committed:
Create static entry page, fix bugs, create containers directory

jbalboni committed:
Create static entry page, fix bugs, create containers directory

jbalboni committed:
Add some tests

jbalboni committed:
Add some tests

jbalboni commented:
This seems to be working now, so it's ready for review, @akainic. Wasn't sure about some of the naming or if I created a Markdown entry page in the right spot.

akainic commented on src/js/edu-benefits/utils/options-for-select.js:
```diff
@@ -0,0 +1,1594 @@
+const countries = [
```
If it looks like these can be shared across apps perhaps we can move it out of the edu-benefits directory and somewhere all apps can easily access? Doesn't have to be done in this first pass though, let's just keep in mind once the HCA is merged in and we can move things around more easily.

jbalboni replied:
Yep, I agree. There are a bunch of hca components that could be shared, too.

akainic commented on src/sass/_hca.scss:
```diff
@@ -0,0 +1,216 @@
+// Variables
```
Was this file not imported in the HCA migration PR? oops, good catch!

jbalboni replied:
Hmm, I didn't mean to change anything here, so I'm not sure why it's showing up in the diff.

akainic commented on test/hca/components/HealthCareApp.spec.jsx:
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
```
I think this was brought in in https://github.com/department-of-veterans-affairs/vets-website/pull/2703/files#diff-a8af7004dab28f2e0d50acf7e7f31b6dR1, just renamed. Maybe don't need this then?

jbalboni replied:
I think I need to revert any changes here, didn't mean to change any hca stuff.

jbalboni committed:
Remove uneeded files

jbalboni committed:
Remove uneeded files

jbalboni committed:
Undoing changes

jbalboni committed:
Undoing changes

jbalboni committed:
Remove unused file

jbalboni committed:
Remove unused file

jbalboni commented:
@akainic I removed a bunch of files that I did not mean to add. Something about the way I pulled in the last round of changes in the hca branch added a bunch of old files. Fixed now.

akainic commented:
Looks great! Let's hold off on merging this until https://github.com/department-of-veterans-affairs/vets-website/pull/2703/ is merged, which should be in the next day or two at the latest.

jbalboni committed:
Merge branch 'edu-benefits-setup-jsb' of github.com:department-of-veterans-affairs/vets-website into edu-benefits-setup-jsb

jbalboni committed:
Fix conflicts

jbalboni committed:
Remove other bad change

jbalboni committed:
Fix linting errors

jbalboni commented:
@akainic I think this is all set now. It's up to date with master and I fixed the linting errors.

jbalboni committed:
Merge branch 'master' into edu-benefits-setup-jsb

jbalboni committed:
Ignore edu benefits apply page in prod

akainic committed:
Merge branch 'master' into edu-benefits-setup-jsb

akainic commented on script/build.js:
```diff
@@ -80,6 +80,7 @@ smith.destination(`../build/${options.buildtype}`);
 const ignoreList = [ 'memorial-benefits/*' ];
 if (options.buildtype === 'production') {
   ignoreList.push('rx/*');
+  ignoreList.push('education/apply-for-education-benefits/apply.md');
```
Should we just have `'education/*'` excluded as Rx does above?

jbalboni replied:
There are some existing static pages under education. I think we'd exclude them if I changed it to that.

akainic replied:
Gotcha, makes sense!

akainic commented:
LGTM with one nit question!

jbalboni committed:
Merge branch 'master' into edu-benefits-setup-jsb
